### PR TITLE
Add security rules

### PR DESCRIPTION
**Pull request by toufik-airane**
I have added security rules, but you can always extends the vanilla config.

#### Edit .eslintrc
* for **vanilla config**
```"extends": "payfit"```

* for **security config**
```"extends": "payfit/security"```

* for both
```"extends": ["payfit", "payfit/security"]```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## eslint-config-payfit
+
+#### Edit .eslintrc
+```
+"extends": [payfit", "payfit/security"],
```
Comment (Kerumen) [inline README.md]:
```suggestion
"extends": ["payfit", "payfit/security"],
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+```
+
+#### Dependencies
+```
+npm info "eslint-config-payfit" peerDependencies
+
+{ 
+"eslint": "^5.3.0",
+"eslint-config-airbnb": "^17.1.0",
```
Comment (Kerumen) [inline README.md]: `eslint-config-airbnb` is a devDep, no need to precise it

File: README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+"eslint-plugin-import": "^2.14.0",
+"eslint-plugin-jsx-a11y": "^6.1.1",
+"eslint-plugin-react': "^7.11.0",
+"eslint-plugin-security": "^1.4.0",
+"eslint-plugin-sonarjs": "^0.3.0",
+"eslint-plugin-no-unsanitized": "^3.0.2",
+"eslint-plugin-xss": "^0.1.9"
```
Comment (Kerumen) [inline README.md]: You lack an indentation on this whole block

File: package.json [hunk 4]
```diff
@@ -27,7 +25,11 @@
     "eslint": "^5.3.0",
     "eslint-plugin-import": "^2.14.0",
     "eslint-plugin-jsx-a11y": "^6.1.1",
-    "eslint-plugin-react": "^7.11.0"
+    "eslint-plugin-react": "^7.11.0",
+    "eslint-plugin-security": "^1.4.0",
+    "eslint-plugin-sonarjs": "^0.3.0",
+    "eslint-plugin-no-unsanitized": "^3.0.2",
```
Comment (Kerumen) [inline package.json]: This one is not well sorted

File: rules/security.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
+module.exports = {
+    plugins: ['sonarjs', 'security', 'no-unsanitized', 'xss'],
+    extends: ['plugin:sonarjs/recommended', 'plugin:security/recommended'],
+    rules: {
+    // PayFit Security Config
```
Comment (Kerumen) [inline rules/security.js]: This comment is useless I guess

File: rules/security.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    },
+    globals: {
+        $env: false,
+        window: true,
+        __DEV__: false,
+        __PROD__: false,
+        __DEBUG__: false,
+        __COVERAGE__: false,
+        __BASENAME__: false,
+    },
```
Comment (Kerumen) [inline rules/security.js]: You can remove `globals` and `env` this is already set by the vanilla config

File: rules/security.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+}
```
Comment (Kerumen) [inline rules/security.js]: You are missing an extra break line

File: rules/base.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  globals: {
+    $env: false,
+    window: true,
+    __DEV__: false,
+    __PROD__: false,
+    __DEBUG__: false,
+    __COVERAGE__: false,
+    __BASENAME__: false,
+  },
+}
```
Comment (Kerumen) [inline rules/base.js]: You are missing an extra break line

File: rules/security.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    'sonarjs/no-duplicated-branches': 'error',
+    'sonarjs/no-identical-functions': 'off',
+    'sonarjs/no-inverted-boolean-check': 'error',
+    'sonarjs/no-redundant-boolean': 'error',
+    'sonarjs/no-small-switch': 'error',
+    'sonarjs/no-useless-catch': 'error',
+    'sonarjs/prefer-immediate-return': 'off',
+    'sonarjs/prefer-object-literal': 'error',
+    'sonarjs/prefer-single-boolean-return': 'error',
+    'sonarjs/prefer-while': 'error',
```
Comment (Kerumen) [inline rules/security.js]: This block is not well indented

Comment (toufik-airane) [reply rules/security.js]: I just indented it with 4 spaces.

File: yarn.lock [hunk 10]
```diff
@@ -1,105 +0,0 @@
-# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (Kerumen) [inline yarn.lock]: You should not delete the `yarn.lock`

File: rules/security.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
+module.exports = {
+    plugins: ['sonarjs', 'security', 'no-unsanitized', 'xss'],
+    extends: ['plugin:sonarjs/recommended', 'plugin:security/recommended'],
+    rules: {
+    // Cross-Site Scripting
+    'xss/no-location-href-assign': 'warn', // in the future, raise to error with specific escape function
```
Comment (Kerumen) [inline rules/security.js]: There is still a problem with the indentation here (or maybe it's GitHub's fault but I don't this so)

File: README.md [hunk 12]
```diff
@@ -0,0 +1,22 @@
+## eslint-config-payfit
+
+#### Edit .eslintrc
+```
+"extends": ["payfit", "payfit/security"],
+```
+
+#### Dependencies
+```
+npm info "eslint-config-payfit" peerDependencies
```
Comment (Kerumen) [inline README.md]: Maybe we can use the same shortcut as `airbnb`:
```
npx install-peerdeps --dev eslint-config-payfit
```

**Comment by toufik-airane** [comment]
Dear @Kerumen,

Are we ready to merge our branch?
Thank you,

**Pull request merged**